File: PracticasProgramacion/Practica_23/python/main.py
"""
Practica_23 - Matriz con puntero al mismo tipo (siguiente_).
"""

from matriz_autoreferenciada import MatrizAutoreferenciada

if __name__ == "__main__":
    a = MatrizAutoreferenciada(2, 2)
    a.llenar(1.0)
    b = MatrizAutoreferenciada(2, 2)
    b.llenar(2.0)
    a.set_siguiente(b)
    c = a.multiplicar_con_siguiente()
    print("A * B (via siguiente):")
    c.imprimir()

File: PracticasProgramacion/Practica_23/cpp/main.cpp
/**
 * Practica_23 - Matrices con puntero/referencia al mismo tipo
 */

#include "MatrizAutoreferenciada.h"
#include <iostream>

int main() {
    MatrizAutoreferenciada A(2, 2);
    A.llenar(1.0);

    MatrizAutoreferenciada B(2, 2);
    B.llenar(2.0);

    A.setSiguiente(&B);
    MatrizAutoreferenciada C = A.multiplicarConSiguiente();
    std::cout << "A * B (via puntero):\n";
    C.imprimir();

    return 0;
}
